Mathlib/CategoryTheory/Adjunction/Mates.lean

```diff
@@ -73,199 +74,565 @@ This can be seen as a bijection of the 2-cells:
-theorem transferNatTrans_counit (f : G ⋙ L₂ ⟶ L₁ ⋙ H) (Y : D) :
-    L₂.map ((transferNatTrans adj₁ adj₂ f).app _) ≫ adj₂.counit.app _ =
-      f.app _ ≫ H.map (adj₁.counit.app Y) := by
-  erw [Functor.map_comp]
-  simp
-#align category_theory.transfer_nat_trans_counit CategoryTheory.transferNatTrans_counit
-
-theorem unit_transferNatTrans (f : G ⋙ L₂ ⟶ L₁ ⋙ H) (X : C) :
-    G.map (adj₁.unit.app X) ≫ (transferNatTrans adj₁ adj₂ f).app _ =
-      adj₂.unit.app _ ≫ R₂.map (f.app _) := by
-  dsimp [transferNatTrans]
-  rw [← adj₂.unit_naturality_assoc, ← R₂.map_comp, ← Functor.comp_map G L₂, f.naturality_assoc,
-    Functor.comp_map, ← H.map_comp]
-  dsimp; simp
-#align category_theory.unit_transfer_nat_trans CategoryTheory.unit_transferNatTrans
-
--- See library note [dsimp, simp]
-end Square
-
-section Self
+@[simps]
+def mateEquiv : (G ⋙ L₂ ⟶ L₁ ⋙ H) ≃ (R₁ ⋙ G ⟶ H ⋙ R₂) where
+  toFun α :=
+    whiskerLeft (R₁ ⋙ G) adj₂.unit ≫
+    whiskerRight (whiskerLeft R₁ α) R₂ ≫
+    whiskerRight adj₁.counit (H ⋙ R₂)
+  invFun β :=
+    whiskerRight adj₁.unit (G ⋙ L₂) ≫
+    whiskerRight (whiskerLeft L₁ β) L₂ ≫
+    whiskerLeft (L₁ ⋙ H) adj₂.counit
+  left_inv α := by
+    ext
+    unfold whiskerRight whiskerLeft
+    simp only [comp_obj, id_obj, Functor.comp_map, comp_app, map_comp, assoc, counit_naturality,
+      counit_naturality_assoc, left_triangle_components_assoc]
+    rw [← assoc, ← Functor.comp_map, α.naturality, Functor.comp_map, assoc, ← H.map_comp,
+      left_triangle_components, map_id]
+    simp only [comp_obj, comp_id]
+  right_inv β := by
+    ext
+    unfold whiskerLeft whiskerRight
+    simp only [comp_obj, id_obj, Functor.comp_map, comp_app, map_comp, assoc,
+      unit_naturality_assoc, right_triangle_components_assoc]
+    rw [← assoc, ← Functor.comp_map, assoc, ← β.naturality, ← assoc, Functor.comp_map,
+      ← G.map_comp, right_triangle_components, map_id, id_comp]
+
+
+/-- A component of a transposed version of the mates correspondence. -/
+theorem mateEquiv_counit (α : G ⋙ L₂ ⟶ L₁ ⋙ H) (d : D) :
+    L₂.map ((mateEquiv adj₁ adj₂ α).app _) ≫ adj₂.counit.app _ =
+      α.app _ ≫ H.map (adj₁.counit.app d) := by
+  erw [Functor.map_comp]; simp
+
+/-- A component of a transposed version of the inverse mates correspondence. -/
+theorem mateEquiv_counit_symm (α : R₁ ⋙ G ⟶ H ⋙ R₂) (d : D) :
+    L₂.map (α.app _) ≫ adj₂.counit.app _ =
+      ((mateEquiv adj₁ adj₂).symm α).app _ ≫ H.map (adj₁.counit.app d) := by
+  conv_lhs => rw [← (mateEquiv adj₁ adj₂).right_inv α]
+  exact (mateEquiv_counit adj₁ adj₂ ((mateEquiv adj₁ adj₂).symm α) d)
+
+/- A component of a transposed version of the mates correspondence. -/
+theorem unit_mateEquiv (α : G ⋙ L₂ ⟶ L₁ ⋙ H) (c : C) :
+    G.map (adj₁.unit.app c) ≫ (mateEquiv adj₁ adj₂ α).app _ =
+      adj₂.unit.app _ ≫ R₂.map (α.app _) := by
+  dsimp [mateEquiv]
+  rw [← adj₂.unit_naturality_assoc]
+  slice_lhs 2 3 =>
+    {
+      rw [← R₂.map_comp, ← Functor.comp_map G L₂]
+      rw [α.naturality]
+    }
```

Comment:
```suggestion
  slice_lhs 2 3 =>
    rw [← R₂.map_comp, ← Functor.comp_map G L₂, α.naturality]
```

Similarly below

Reply:
This syntax also works for multiple line tactic proofs

Reply:
Thanks @dagurtomas , I've propagated this suggestion throughout the file.